### ISSUE TITLE
`poetry install` console output corrupted on Windows

### ISSUE THREAD

**Issue by jimbo1qaz** [sdispater/poetry#1025]
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [ ] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Windows 10 x64
- **Poetry version**: 0.12.11
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: https://gist.github.com/jimbo1qaz/d97d3da78cece9a38157656f9a082034

## Issue

When I run `poetry install`, and look at console output, all upgrade lines have the "arrow and new version" overwriting the beginning of the line of text.

```
 -> 2019.3.9)certifi (2018.11.29
  - Installing makefun (1.6.1)
 -> 6.0.0)ng more-itertools (4.3.0
 -> 0.9.0)ng pluggy (0.8.1
 -> 1.8.0)ng py (1.7.0
 -> 4.5.3)ng coverage (4.5.2
  - Installing decopatch (1.4.1)
 -> 1.16.2)g numpy (1.16.1
 -> 4.19.15) pyqt5-sip (4.19.13
 -> 4.3.1)ng pytest (4.2.0
 -> 2.8.0)ng python-dateutil (2.7.5
 -> 3.0.3)ng matplotlib (3.0.2
 -> 3.5.dev0+g5bed1af2 5bed1af)
 -> 5.12.1)g pyqt5 (5.11.3
  - Installing pytest-cases (1.5.0)
 -> 1.10.2)g pytest-mock (1.10.0
 -> 0.15.89) ruamel.yaml (0.15.87
  - Installing corrscope (0.5.0-pre)
```

This happens in Pycharm, Appveyor, and cmd, and appears to be there since 0.12.0 or earlier.

**Comment by jimbo1qaz** [comment]
I think poetry prints a stray carriage return on Windows. I theorize poetry strips the final character using `[:-1]` (which removes the newline on Unix and universal newlines, but when performed Windows binary output `\r\n`, leaves behind `\r`.)

poetry/installation/installer.py
```py
            self._io.writeln(
                "  - Updating <info>{}</> (<comment>{}</> -> <comment>{}</>)".format(
                    target.pretty_name,
                    source.full_pretty_version,
                    target.full_pretty_version,
                )
            )
```

Is `source` obtained by parsing CLI output from `pip list` or similar?

**Comment by Pluckerpluck** [comment]
I'm not sure that `universal_newlines` does what you expect it to. What should be done is that `freeze_output.split(os.linesep)`.

At least, that's my guess based off a wild assumption that windows is presenting the `\r\n` newline rather than it being some more niche bug. I haven't looked at the code at all yet.

**Comment by KyleKing** [comment]
Looks like this is fixed in 1.0.0b5!

**Comment by finswimmer** [comment]
Thanks for reporting.

I close this issue.
